crates/anvil/src/eth/pool/mod.rs

```diff
@@ -342,6 +347,31 @@ impl PoolInner {
 
         removed
     }
+
+    /// Remove transactions by sender address
+    pub fn remove_transactions_by_address(&mut self, sender: Address) -> Vec<Arc<PoolTransaction>> {
+        let mut tx_hashes: Vec<TxHash> =
+            self.pending_transactions.transactions_by_sender(sender).map(|tx| tx.hash()).collect();
+        tx_hashes.extend(
+            self.ready_transactions
+                .transactions_by_sender(sender)
+                .map(|tx| tx.hash())
+                .collect::<Vec<TxHash>>(),
+        );
+
+        if tx_hashes.is_empty() {
+            return vec![]
+        }
```

Comment:
can we extract this to a `transactions_by_sender` that merges those?

Reply:
Refactored! Let me know if this is what you meant!